Failed to build with SPGDEBUG

@LecrisUT 

It may be my silly misunderstanding, but cmake build with the following commands 
```shell
cmake -S . -B build -D CMAKE_BUILD_TYPE=Debug
cmake --build build
```
shows the following messages.
```
/Users/kshinohara/ghq/github.com/lan496/spglib/src/debug.h:49:20: warning: inline function 'warning_memory' is not defined [-Wundefined-inline]
extern inline void warning_memory(char const* what);
                   ^
/Users/kshinohara/ghq/github.com/lan496/spglib/src/kpoint.c:124:9: note: used here
        warning_memory("dense_ir_mapping_table");
        ^
1 warning generated.
```

I thought `CMAKE_BUILD_TYPE=Debug` set `SPGLIB_DEBUG`, and it included the debug functions.
How to compile with the debug functions?

After specifying `-DCMAKE_BUILD_TYPE=Debug`, I still see the warnings like
`/Users/kshinohara/ghq/github.com/lan496/spglib/src/debug.h:49:20: warning: inline function 'warning_memory' is not defined [-Wundefined-inline]`
```
❯ cmake -B build -DCMAKE_BUILD_TYPE=Debug
-- Found Git: /opt/homebrew/bin/git (found version "2.45.0")
-- The C compiler identification is AppleClang 15.0.0.15000309
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Spglib:: Build type: Debug
-- Spglib:: Version 2.5.1.dev2+g34723dd
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found Python: /Users/kshinohara/.pyenv/versions/3.11.9/envs/spglib/bin/python3.11 (found version "3.11.9") found components: Interpreter
-- Configuring done (6.0s)
-- Generating done (0.1s)
-- Build files have been written to: /Users/kshinohara/ghq/github.com/lan496/spglib/build
```

--- a/src/debug.c
+++ b/src/debug.c
@@ -108,18 +108,6 @@ void debug_print(char const* format, ...) {
     vfprintf(stdout, format, argptr);
     va_end(argptr);
 }
-#else
-void debug_print_matrix_d3(double const a[3][3]) {}
-
-void debug_print_matrix_i3(int const a[3][3]) {}
-
-void debug_print_vectors_d3(double const a[][3], int size) {}
-
-void debug_print_vector_d3(double const a[3]) {}
-
-void debug_print_vectors_with_label(double const a[][3], int const b[],
-                                    int size) {}
-void debug_print(char const* format, ...) {}
 #endif
 
 #ifdef SPGWARNING
@@ -130,8 +118,6 @@ void warning_print(char const* format, ...) {
     vfprintf(stderr, format, argptr);
     va_end(argptr);
 }
-#else
-void warning_print(char const* format, ...) {}
 #endif
 
 #ifdef SPGINFO
@@ -142,10 +128,4 @@ void info_print(char const* format, ...) {
     vfprintf(stderr, format, argptr);
     va_end(argptr);
 }
-#else
-void info_print(char const* format, ...) {}
 #endif
-
-void warning_memory(char const* what) {
-    warning_print("Spglib: Memory could not be allocated: %s\n", what);
-}

--- a/src/debug.h
+++ b/src/debug.h
@@ -34,16 +34,40 @@
 
 #pragma once
 
+// If printing is disabled define empty inlined functions to be optimized out
+#ifndef SPGDEBUG
+static inline void debug_print_matrix_d3(double const a[3][3]) {}
+
+static inline void debug_print_matrix_i3(int const a[3][3]) {}
+
+static inline void debug_print_vectors_d3(double const a[][3], int size) {}
+
+static inline void debug_print_vector_d3(double const a[3]) {}
+
+static inline void debug_print_vectors_with_label(double const a[][3],
+                                                  int const b[], int size) {}
+static inline void debug_print(char const* format, ...) {}
+#endif
+#ifndef SPGWARNING
+static inline void warning_print(char const* format, ...) {}
+#endif
+#ifndef SPGINFO
+static inline void info_print(char const* format, ...) {}
+#endif
+
+// Otherwise the print functions are proper functions
 // Main print interface
-extern inline void warning_print(char const* format, ...);
-extern inline void debug_print(char const* format, ...);
-extern inline void info_print(char const* format, ...);
-extern inline void debug_print_matrix_d3(double const a[3][3]);
-extern inline void debug_print_matrix_i3(int const a[3][3]);
-extern inline void debug_print_vector_d3(double const a[3]);
-extern inline void debug_print_vectors_d3(double const a[][3], int size);
-extern inline void debug_print_vectors_with_label(double const a[][3],
-                                                  int const b[], int size);
+void warning_print(char const* format, ...);
+void debug_print(char const* format, ...);
+void info_print(char const* format, ...);
+void debug_print_matrix_d3(double const a[3][3]);
+void debug_print_matrix_i3(int const a[3][3]);
+void debug_print_vector_d3(double const a[3]);
+void debug_print_vectors_d3(double const a[][3], int size);
+void debug_print_vectors_with_label(double const a[][3], int const b[],
+                                    int size);
 
 // Common messages
-extern inline void warning_memory(char const* what);
+inline void warning_memory(char const* what) {
+    warning_print("Spglib: Memory could not be allocated: %s\n", what);
+}

